components/cdc/src/endpoint.rs

```diff
@@ -306,6 +319,47 @@ impl ResolvedRegionHeap {
         self.clear();
         self.heap.shrink_to(min_capacity);
     }
+
+    // extreme outier match the following two conditions:
+    // 1. https://en.wikipedia.org/wiki/Box_plot
+    // 2. the gap with min_ts is larger than RAW_RESOLVED_TS_OUTLIER_THRESHOLD.
+    fn get_extreme_outlier(&self, min_ts: TimeStamp) -> Vec<u64> {
+        if self.heap.is_empty() {
+            return vec![];
+        } else if self.heap.len() < 4 {
+            // if count is smaller than 4, just return one outlier if exists.
+            let resolved = self.heap.peek().unwrap().0.resolved_ts;
+            if Duration::from_millis(min_ts.physical().saturating_sub(resolved.physical()))
+                > RAW_RESOLVED_TS_OUTLIER_THRESHOLD
+            {
+                return vec![self.heap.peek().unwrap().0.region_id];
+            }
+        } else {
+            let ts_vec: Vec<TimeStamp> = self.heap.iter().map(|iter| iter.0.resolved_ts).collect();
```

Comment:
It would be more efficient to gather all raw resolved-ts and sort, than use a heap.

Reply:
yes, `ResolvedRegionHeap` is reused to store raw region resolved_ts info here.

Reply:
Added ResolvedRegionVec to improve efficiency.